skynet/skynet.go

```diff
@@ -39,19 +49,29 @@ func (c *Client) Upload(namespace, digest string, content []byte, headers ...sky
 
 	data[digest] = buf
 
-	return c.skynet.Upload(data, opts, headers...)
+	skylink, err := c.skynet.Upload(data, opts)
+	if err != nil {
+		return "", err
+	}
+
+	if pin {
+		time.Sleep(time.Second * 10)
```

Comment:
@jay-dee7 , is the sleep really needed here?

Reply:
i tried removing it and it works fine now. marking it as resolved